gpii/node_modules/flowManager/src/CloudBasedFlowManager.js

```diff
@@ -119,20 +119,35 @@ fluid.defaults("gpii.flowManager.cloudBased.oauth2", {
             }
         },
         userService: {
-            type: "gpii.oauth2.userService",
-            options: {
-                components: {
-                    dataStore: "{gpii.oauth2.dataStore}"
-                }
-            }
+            type: "gpii.oauth2.userService"
+        },
+        authGrantFinder: {
+            type: "gpii.oauth2.authGrantFinder"
+        }
+    },
+    distributeOptions: {
+        setDataStoreOnAuthGrantFinder: {
+            record: "{gpii.oauth2.dataStore}",
+            target: "{that > authGrantFinder}.options.components.authorizationService.options.components.dataStore"
```

Comment:
This is a terrifyingly deep target path, as well as placing unncessary conditions on the authGrantFinder position. Better would be {that gpii.oauth2.authorizationService}.options.components.dataStore